src/frontend/src/scheduler/distributed/query.rs

```diff
@@ -374,6 +384,85 @@ impl QueryRunner {
             stage_execution.stop().await;
         }
     }
+
+    fn create_plan_fragment(&self) -> PlanFragment {
+        let root_stage_id = self.query.stage_graph.root_stage_id;
+        let root_stage = self.stage_executions.get(&root_stage_id).unwrap();
+        let plan_node_prost =
+            Self::convert_plan_node(root_stage, &root_stage.stage.root, ROOT_TASK_ID);
+        let exchange_info = root_stage.stage.exchange_info.clone();
+        PlanFragment {
+            root: Some(plan_node_prost),
+            exchange_info: Some(exchange_info),
+        }
+    }
+
+    fn convert_plan_node(
+        stage_execution: &StageExecution,
+        execution_plan_node: &ExecutionPlanNode,
+        task_id: TaskId,
```

Comment:
These codes are copied from `StageExecution`?

Reply:
I think you mean LocalQueryExecution? 
Yes the code are copied, basically extract the root fragment from plan. i remembered that there are some logic different so no-reuse